pertpy/tools/_metrics_3g.py

```diff
@@ -0,0 +1,200 @@
+from collections.abc import Mapping
+from functools import partial
+from types import MappingProxyType
+from typing import TYPE_CHECKING, Any, Literal, Optional
+
+import anndata as ad
+import numpy as np
+import pandas as pd
+import pynndescent
+import scanpy as sc
+from scipy.sparse import issparse
+from scipy.sparse import vstack as sp_vstack
+from sklearn.base import ClassifierMixin
+from sklearn.linear_model import LogisticRegression
+
+from pertpy.tools._distances._distances import Distance
+
+if TYPE_CHECKING:
+    from numpy.typing import NDArray
+
+
+def compare_de(X: np.ndarray, Y: np.ndarray, C: np.ndarray, shared_top: int = 100, **kwargs) -> dict:
+    """Compare DEG across real and simulated perturbations.
+
+    Computes DEG for real and simulated perturbations vs. control and calculates
+    metrics to evaluate similarity of the results.
+
+    Args:
+        X: Real perturbed data.
+        Y: Simulated perturbed data.
+        C: Control data
+        shared_top: The number of top DEG to compute the proportion of their intersection.
+        **kwargs: arguments for `scanpy.tl.rank_genes_groups`.
+    """
+    n_vars = X.shape[1]
+    assert n_vars == Y.shape[1] == C.shape[1]
+
+    shared_top = min(shared_top, n_vars)
+    vars_ranks = np.arange(1, n_vars + 1)
+
+    adatas_xy = {}
+    adatas_xy["x"] = ad.AnnData(X, obs={"label": "comp"})
+    adatas_xy["y"] = ad.AnnData(Y, obs={"label": "comp"})
+    adata_c = ad.AnnData(C, obs={"label": "ctrl"})
+
+    results = pd.DataFrame(index=adata_c.var_names)
+    top_names = []
+    for group in ("x", "y"):
+        adata_joint = ad.concat((adatas_xy[group], adata_c), index_unique="-")
+
+        sc.tl.rank_genes_groups(adata_joint, groupby="label", reference="ctrl", key_added="de", **kwargs)
+
+        srt_idx = np.argsort(adata_joint.uns["de"]["names"]["comp"])
+        results[f"scores_{group}"] = adata_joint.uns["de"]["scores"]["comp"][srt_idx]
+        results[f"pvals_adj_{group}"] = adata_joint.uns["de"]["pvals_adj"]["comp"][srt_idx]
+        # needed to avoid checking rankby_abs
+        results[f"ranks_{group}"] = vars_ranks[srt_idx]
+
+        top_names.append(adata_joint.uns["de"]["names"]["comp"][:shared_top])
+
+    metrics = {}
+    metrics["shared_top_genes"] = len(set(top_names[0]).intersection(top_names[1])) / shared_top
+    metrics["scores_corr"] = results["scores_x"].corr(results["scores_y"], method="pearson")
+    metrics["pvals_adj_corr"] = results["pvals_adj_x"].corr(results["pvals_adj_y"], method="pearson")
+    metrics["scores_ranks_corr"] = results["ranks_x"].corr(results["ranks_y"], method="spearman")
+
+    return metrics
+
+
+def compare_class(X: np.ndarray, Y: np.ndarray, C: np.ndarray, clf: Optional[ClassifierMixin] = None) -> float:
```

Comment:
Sounds a bit like Augur tbh? Augur is already implemented in pertpy. Not sure whether we need this...